docs/rtk-query/usage-with-typescript.mdx

```diff
@@ -497,3 +497,184 @@ function MaybePost({ id }: { id?: number }) {
+that error will be transformed into a `SerializedError` shape. Users should ensure that they are checking which
+kind of error they are dealing with prior to attempting to access it's properties. This can be done in a type safe manner either
+by using a type guard, e.g. by checking for [discriminated properties](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing),
+or using a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates).
+
+When using [`fetchBaseQuery`](./api/fetchBaseQuery.mdx), as your base query,
+errors will be of type `FetchBaseQueryError | SerializedError`. The specific shapes of those types can be seen below.
+
+```ts title="FetchBaseQueryError type"
+export type FetchBaseQueryError =
+  | {
+      /**
+       * * `number`:
+       *   HTTP status code
+       */
+      status: number
+      data: unknown
+    }
+  | {
+      /**
+       * * `"FETCH_ERROR"`:
+       *   An error that occurred during execution of `fetch` or the `fetchFn` callback option
+       **/
+      status: 'FETCH_ERROR'
+      data?: undefined
+      error: string
+    }
+  | {
+      /**
+       * * `"PARSING_ERROR"`:
+       *   An error happened during parsing.
+       *   Most likely a non-JSON-response was returned with the default `responseHandler` "JSON",
+       *   or an error occurred while executing a custom `responseHandler`.
+       **/
+      status: 'PARSING_ERROR'
+      originalStatus: number
+      data: string
+      error: string
+    }
+  | {
+      /**
+       * * `"CUSTOM_ERROR"`:
+       *   A custom error type that you can return from your `fetchFn` where another error might not make sense.
+       **/
+      status: 'CUSTOM_ERROR'
+      data?: unknown
+      error: string
+    }
+```
+
+```ts title="SerializedError type"
+export interface SerializedError {
+  name?: string
+  message?: string
+  stack?: string
+  code?: string
+}
+```
+
+### Error result example
+
+When using `fetchBaseQuery`, the `error` property returned from a hook will have the type `FetchBaseQueryError | SerializedError | undefined`.
+If an error is present, you can access error properties after narrowing the type to either `FetchBaseQueryError` or `SerializedError`.
+
+```tsx
+import { api } from './services/api'
+
+function PostDetail() {
+  const { data, error, isLoading } = usePostsQuery()
+
+  if (isLoading) {
+    return <div>Loading...</div>
+  }
+
+  if (error) {
+    if ('status' in error) {
+      // you can access all properties of `FetchBaseQueryError` here
+      const errMsg = 'error' in err ? err.error : JSON.stringify(err.data)
+
+      return (
+        <div>
+          <div>An error has occurred:</div>
+          <div>{errMsg}</div>
+        </div>
+      )
+    }
+
+    // you can access all properties of `SerializedError` here
+    return <div>{error.message}</div>
```

Comment:
```suggestion
    else {
        // you can access all properties of `SerializedError` here
        return <div>{error.message}</div>
    }
```
technically not necessary, but I'd add it for clarity - people might skim over the `return` above.